src/providers/wms/qgsxyzconnectiondialog.cpp

```diff
@@ -44,3 +44,27 @@ QgsXyzConnection QgsXyzConnectionDialog::connection() const
     conn.zMax = mSpinZMax->value();
   return conn;
 }
+
+void QgsXyzConnectionDialog::on_mCheckBoxZMin_stateChanged( int state )
```

Comment:
We're moving away from using these Qt autoconnections - they are really fragile and break easily. See https://lists.osgeo.org/pipermail/qgis-developer/2017-February/047288.html for some of the reasons these are hated.

You should be able to get away with the dedicated slots, by using something like

    connect( mCheckBoxZMin, &QCheckBox::toggled, mSpinZMin, QSpinBox::setEnabled )

(and similar for mCheckBoxZMax)

This approach has the added benefit that breaking changes will cause compilation to fail, making the changes robust for future QGIS releases.